pyccel/codegen/printing/ccode.py

```diff
@@ -1146,45 +1144,117 @@ def _print_Assign(self, expr):
         if isinstance(expr.rhs, FunctionCall) and isinstance(expr.rhs.dtype, NativeTuple):
             self._temporary_args = [VariableAddress(a) for a in expr.lhs]
             return '{};'.format(self._print(expr.rhs))
+        if isinstance(expr.rhs, (NumpyArray)):
+            return self.copy_NumpyArray_Data(expr)
+        if isinstance(expr.rhs, (NumpyFull)):
+            return self.arrayFill(expr)
+        if isinstance(expr.rhs, NumpyArange):
+            return self.print_NumpyArange(expr.rhs, expr.lhs)
```

Comment:
I really should have made the same comment on @MenoIy 's PR. :/

Reply:
> _EmilyBourne_
> Functions in pyccel always follow the rule that if their name is _print_X then the argument expr is of type X. Not following this rule is dangerous as this function may be called when you do not want it to be. For example:
> ```
> a = np.array([1,2,3])*4
> ```
> Please rename this function
>  
> _saniQQ_
> I need the Assign expression to be passed to the function if i add the _ the printer will treat it as a rhs and i won't have acess to the lhs
> 
> _EmilyBourne_
> Ah sorry! I missed the fact that the _ is missing. In which case you are right, this is not a dangerous case, but I don't think it is very readable as it is easy to miss that and then be surprised that the argument is not the expected type. If you could find a clearer name that would be great. In any case it needs a doc string


> Please add a docstring to this function. Codacy doesn't complain in this file as all the _print_X functions would have nearly the same docstring, but any other functions should be properly documented

Reply:
ah sorry, @saniQQ  can you please rename the function?
```Python
        """ print the assignment of a NumpyArange
        parameters
        ----------
            expr : NumpyArange
                The node holding NumpyArange
            lhs : Variable
                 The left hand of Assign
        Return
        ------
            String
                Return string that contains the Assign code and the For loop
                responsible for filling the array values
        """
```
feel free to update this doc
and for the name you can name it fill_NumpyArange or you can pick a better name